packages/backend-app-api/src/wiring/BackendInitializer.ts

```diff
@@ -407,6 +407,26 @@ export class BackendInitializer {
       // The startup failed, but we may still want to do cleanup so we continue silently
     }
 
+    // Get all plugins.
+    const pluginMap = new Map<string, boolean>();
+    for (const feature of this.#registrations) {
+      for (const r of feature.getRegistrations()) {
+        if (r.type === 'plugin') {
+          pluginMap.set(r.pluginId, true);
+        }
+      }
+    }
+    const allPluginIds = Array.from(pluginMap.keys());
+
+    // Iterate through all plugins and run their shutdown hooks.
+    await Promise.allSettled(
+      allPluginIds.map(async pluginId => {
+        const lifecycleService = await this.#getPluginLifecycleImpl(pluginId);
+        await lifecycleService.shutdown();
+      }),
+    );
```

Comment:
Thinking it could be useful to gather and log any errors here for visibility

Reply:
The logging logic lives in the plugin lifecycle `shutdown` method.  That work for now?